features/tailormade_profile/src/main/java/com/future/tailormade_profile/feature/editProfile/view/EditProfileFragment.kt

```diff
@@ -0,0 +1,144 @@
+package com.future.tailormade_profile.feature.editProfile.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.viewModelScope
+import com.future.tailormade.base.view.BaseFragment
+import com.future.tailormade.config.Constants
+import com.future.tailormade.util.extension.debounceOnTextChanged
+import com.future.tailormade.util.extension.isPhoneNumberValid
+import com.future.tailormade.util.extension.toDateString
+import com.future.tailormade_profile.R
+import com.future.tailormade_profile.databinding.FragmentEditProfileBinding
+import com.future.tailormade_profile.feature.editProfile.viewModel.EditProfileViewModel
+import com.google.android.material.datepicker.MaterialDatePicker
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
+
+@AndroidEntryPoint
+class EditProfileFragment : BaseFragment() {
+
+  private val viewModel: EditProfileViewModel by viewModels()
+
+  private var birthDate: Long = 0L
+
+  private lateinit var binding: FragmentEditProfileBinding
+
+  private lateinit var birthDatePicker: MaterialDatePicker<Long>
+
+  override fun getLogName(): String =
+      "com.future.tailormade_profile.feature.editProfile.view.EditProfileFragment"
+
+  override fun getScreenName(): String = "Edit Profile"
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+      savedInstanceState: Bundle?): View? {
+    setupDatePicker()
+    binding = FragmentEditProfileBinding.inflate(inflater, container, false)
+
+    with(binding) {
+      buttonSubmitEditProfileForm.setOnClickListener {
+        submitForm(editTextNameEditProfile.text.toString(),
+            editTextPhoneNumberEditProfile.text.toString(),
+            editTextBirthDateEditProfile.text.toString(),
+            editTextLocationEditProfile.text.toString())
+      }
+
+      editTextLocationEditProfile.debounceOnTextChanged(
+          viewModel.viewModelScope, viewModel::updateLocations)
+
+      textInputBirthDateEditProfile.setEndIconOnClickListener {
+        showDatePicker()
+      }
+    }
+
+    setupProfileDataObserver()
+    setupLocationObserver()
```

Comment:
do observe live data in onActivityCreated in fragment
https://proandroiddev.com/5-common-mistakes-when-using-architecture-components-403e9899f4cb